language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -42,27 +43,38 @@ def pretty_format_yaml(argv=None):
     # Prevent ruamel.yaml to wrap yaml lines
     yaml.width = maxsize
 
+    separator = '---\n'
+
     for yaml_file in set(args.filenames):
         with open(yaml_file) as f:
             string_content = ''.join(f.readlines())
 
-        try:
-            content = yaml.load(string_content)
-
-            if not isinstance(content, (list, dict)):
-                # skip files containing primitive types (unstructured text)
-                continue
+        separator_pattern = '^--- ?.*\\n'
+        original_docs = re.split(separator_pattern, string_content, flags=re.MULTILINE)
+        pretty_docs = []
 
-            pretty_content = StringIO()
-            yaml.dump(content, pretty_content)
-
-            if string_content != pretty_content.getvalue():
+        try:
+            for doc in original_docs:
```

Comment:
Please use `load_all` from `ruamel.yaml` (as proposed in https://github.com/macisamuele/language-formatters-pre-commit-hooks/pull/3/#discussion_r418655126)

Reply:
As I mentioned, it isn't feasible because of what it does with newlines. My current approach seems to work fine. It uses load for everything except detecting the doc separators.